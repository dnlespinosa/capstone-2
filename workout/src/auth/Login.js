import React, { useContext, useState } from 'react';
import { useNavigate } from 'react-router-dom';

const Login = ({ login }) => {
    const history  = useNavigate()
    const [formData, setFormData] = useState({
        username: '', 
        password: ''
    })

    async function handleSubmit(e) {
        e.preventDefault();
        let res = await login(formData)
        if (res.success) {
            history('/')
        } else {
            console.log(res.errors)
        }
    }

    function handleChange(evt) {
        const { name, value } = evt.target;
        setFormData(l => ({ ...l, [name]: value}))
    }

    return (
        <>
            <h2>Login Form</h2>

            <form onSubmit={handleSubmit}>
                <label>Username</label>
                <input 
                    name='username'
                    value={formData.username}
                    onChange={handleChange} 
                />

                <label>Password</label>
                <input 
                    name='password'
                    value={formData.password}
                    onChange={handleChange} 
                />

                <button type='submit' onSubmit={handleSubmit}>Login!</button>
            </form>
        </>
    )
}

export default Login