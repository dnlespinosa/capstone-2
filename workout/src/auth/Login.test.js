import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import Login from "./Login";
import { UserProvider } from "../testUtils";


it("renders without crashing", function () {
    render(
        <MemoryRouter>
          <UserProvider>
            <Login />
          </UserProvider>
        </MemoryRouter>,
    );
  });