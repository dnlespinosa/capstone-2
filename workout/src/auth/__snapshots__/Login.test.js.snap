// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    style="color: white;"
  >
    <h2>
      Login Form
    </h2>
    <form>
      <label>
        Username
      </label>
      <input
        name="username"
        value=""
      />
      <label>
        Password
      </label>
      <input
        name="password"
        value=""
      />
      <button
        type="submit"
      >
        Login!
      </button>
    </form>
  </div>
</DocumentFragment>
`;
